src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -86,18 +86,22 @@ internal abstract class PENamedTypeSymbol : NamedTypeSymbol
         private static readonly UncommonProperties s_noUncommonProperties = new UncommonProperties();
         private UncommonProperties _lazyUncommonProperties;
 
-        private UncommonProperties GetUncommonProperties()
+        private UncommonProperties GetUncommonProperties(bool force = false)
         {
             var result = _lazyUncommonProperties;
             if (result != null)
             {
 #if DEBUG
                 Debug.Assert(result != s_noUncommonProperties || result.IsDefaultValue(), "default value was modified");
 #endif
-                return result;
+                // Avoid returning the default instance when caller passes 'force: true'.
+                if (!force || result != s_noUncommonProperties)
+                {
+                    return result;
+                }
             }
 
-            if (this.IsUncommon())
+            if (force || this.IsUncommon())
             {
                 result = new UncommonProperties();
                 return Interlocked.CompareExchange(ref _lazyUncommonProperties, result, null) ?? result;
```

Comment:
this may still be incorrect. 'null' should perhaps be replaced with value of 'result' before line 106.

It's actually a bit tricky to test this, because IIRC when we try inserting a sequence of calls like `GetUncommonProperties(force: false); GetUncommonProperties(force: true);` we end up hitting some error in the first call due to incomplete initialization of the symbol

Reply:
I ended up just adding an assert here to limit the way the 'force' parameter can be used. I was concerned that being able to observe different instances when calling `GetUncommonProperties(force: false)` multiple times could be error prone in its own way.

The whole mismatch here seems to result from the fact that the existing properties on `UncommonProperties` are meant to be initialized lazily, but it's more convenient here to initialize the properties eagerly.